SneakPeek/src/com/cyscorpions/dalejulian/sneakpeek/activities/SneakerDetailsActivity.java

```diff
@@ -0,0 +1,138 @@
+		setContentView(R.layout.sneaker_entry_activity);
+		sneakerId = getIntent().getStringExtra(SneakerEntryListActivity.LIST_ID_EXTRA);
+		Log.i("Sneaker Entry", sneakerId);
+		findResources();
+		setupContentByExtras();
+		getActionBar().setDisplayHomeAsUpEnabled(true);
+	}
+
+	private void findResources() {
+		mName = (TextView) findViewById(R.id.txtSneakerName);
+		mBrand = (TextView) findViewById(R.id.txtSneakerBrand);
+		mSellValue = (TextView) findViewById(R.id.txtSellingValue);
+		mRarity = (TextView) findViewById(R.id.txtSneakerRarity);
+		mCategory = (TextView) findViewById(R.id.txtSneakerCategory);
+		mDesc = (TextView) findViewById(R.id.txtSneakerDescription);
+		mThumbnail = (ImageView) findViewById(R.id.imgSneakerEntry);
+	}
+
+	private void setupContentByExtras() {
+
+		mName.setText(getIntent().getStringExtra(
+				SneakerEntryListActivity.LIST_TITLE_EXTRA));
+		mBrand.setText(getIntent().getStringExtra(
+				SneakerEntryListActivity.LIST_BRAND_EXTRA));
+		mSellValue.setText(getIntent().getStringExtra(
+				SneakerEntryListActivity.LIST_SELLING_EXTRA));
+		mRarity.setText(getIntent().getStringExtra(
+				SneakerEntryListActivity.LIST_RARITY_EXTRA));
+		mCategory.setText(getIntent().getStringExtra(
+				SneakerEntryListActivity.LIST_CATEGORY_EXTRA));
+		mDesc.setText(getIntent().getStringExtra(
+				SneakerEntryListActivity.LIST_DESC_EXTRA));
+
+		@SuppressWarnings("deprecation")
+		Drawable imageDrawable = getResources().getDrawable(
+				getIntent().getIntExtra(SneakerEntryListActivity.LIST_IMGID_EXTRA, 0));
+		mThumbnail.setImageDrawable(imageDrawable);
+		sneaker.setName(mName.getText().toString());
+		sneaker.setBrand(mBrand.getText().toString());
+		sneaker.setSellingValue(mSellValue.getText().toString());
+		sneaker.setDescription(mDesc.getText().toString());
+		sneaker.setThumbnailId(getIntent().getIntExtra(
+				SneakerEntryListActivity.LIST_IMGID_EXTRA, 0));
+		sneaker.setRarity(mRarity.getText().toString());
+		sneaker.setCategory(mCategory.getText().toString());
+	}
+
+	@Override
+	public boolean onOptionsItemSelected(MenuItem item) {
+		int id = item.getItemId();
+		if (id == R.id.action_settings) {
+
+			return true;
+		}
+
+		if (item.getTitle() == "Edit details") {
+			Intent i = new Intent(SneakerDetailsActivity.this,
+					EditSneakerEntryActivity.class);
+			i.putExtra(DETAILS_NAME_EXTRA, sneaker.getName());
+			i.putExtra(DETAILS_BRAND_EXTRA, sneaker.getBrand());
+			i.putExtra(DETAILS_SELLING_EXTRA, sneaker.getSellingValue());
+			i.putExtra(DETAILS_RARITY_EXTRA, sneaker.getRarity());
+			i.putExtra(DETAILS_CATEGORY_EXTRA, sneaker.getCategory().getName()
+					.toString());
+			i.putExtra(DETAILS_DESC_EXTRA, sneaker.getDescription());
+			i.putExtra(DETAILS_IMGID_EXTRA, sneaker.getThumbnailId());
+			i.putExtra(DETAILS_TITLE_EXTRA, sneaker.getTitleName());
+			i.putExtra(DETAILS_UUID_EXTRA, sneakerId);
+			i.putExtra(DETAILS_FROMDETAILS_EXTRA, true);
+			startActivityForResult(i, 1);
```

Comment:
Implement a requestCode in the EditSneakerEntryActivity class and statically get it from there